fictadvisor-api/src/v2/api/services/ScheduleService.ts

```diff
@@ -465,19 +520,29 @@ export class ScheduleService {
       startTime.setMinutes(durationTime.startTime.getMinutes());
     }
 
-    const endTime = period === Period.NO_PERIOD
-      ? durationTime.endTime
-      : await this.getLastEndDate(durationTime.endTime ?? event.endTime, period);
+    const eventsAmount = await this.calculateEventsAmount(startTime, period);
 
-    const eventInfoForUpdate = await this.getEventInfoForUpdate(period, startTime, endTime, week, eventInfo, event);
+    let teacherForceChanges = false;
+    if (teachers && disciplineId && !event.teacherForceChanges) {
+      teacherForceChanges = await this.validateTeachersDiscipline(teachers, disciplineId);
+    }
+
+    if (eventInfo) {
+      const index = await this.getIndexOfLesson(week, event);
```

Comment:
Add null check for index and create common variable index for if else blocks to avoid code duplication